### DISCUSSION TITLE
拿到account 信息后是否本地一直有效？还是必须受小米公司控制，定期去站点更新？

### DISCUSSION THREAD

**Discussion by snowyu** [idootop/mi-gpt#172] [Q&A] [answered]
as title

**Comment by idootop** [comment, marked as answer]
如果一直在运行的话，会自动刷新 token。account 登录凭证肯定是有时效的。

**Reply by snowyu** [reply]
也就是小米一旦封杀,或者停止服务,这个设备也就废铁一堆了,和京东音箱一样.设备控制权不在用户手上,
现在的所谓的智能音箱,全是服务,就没一个是真正的离线可用的产品. 还好现在有了开源LLM.

**Reply by idootop** [reply]
是的。不过你也可以刷机破解他们的硬件，把控制权掌控在自己手里。准确地讲它就是个扬声器+几个麦克风。

**Reply by idootop** [reply]
或者你也可以用开发版或废旧的二手手机自己 DIY，网上也有好多类似的开源项目和硬件可以参考。
